src/SFML/Window/Win32/JoystickImpl.cpp

```diff
@@ -398,35 +402,31 @@ JoystickState JoystickImpl::update()
 void JoystickImpl::initializeDInput()
 {
     // Try to load dinput8.dll
-    dinput8dll = LoadLibraryA("dinput8.dll");
+    auto localDinput8LibraryPtr = std::make_unique<LibraryLoader>(L"dinput8.dll");
 
-    if (dinput8dll)
-    {
-        // Try to get the address of the DirectInput8Create entry point
-        typedef HRESULT(WINAPI *DirectInput8CreateFunc)(HINSTANCE, DWORD, REFIID, LPVOID*, LPUNKNOWN);
-        DirectInput8CreateFunc directInput8Create = reinterpret_cast<DirectInput8CreateFunc>(reinterpret_cast<void*>(GetProcAddress(dinput8dll, "DirectInput8Create")));
+    if (!localDinput8LibraryPtr->getIsLoaded())
+        return;
 
-        if (directInput8Create)
-        {
-            // Try to acquire a DirectInput 8.x interface
-            HRESULT result = directInput8Create(GetModuleHandleW(nullptr), 0x0800, guids::IID_IDirectInput8W, reinterpret_cast<void**>(&directInput), nullptr);
+    // Try to get the address of the DirectInput8Create entry point
+    typedef HRESULT(WINAPI *DirectInput8CreateFunc)(HINSTANCE, DWORD, REFIID, LPVOID*, LPUNKNOWN);
```

Comment:
Since this is C++11, how about `using` declarations instead of `typedef`'s?

```suggestion
    using DirectInput8CreateFunc = HRESULT (WINAPI*)(HINSTANCE, DWORD, REFIID, LPVOID*, LPUNKNOWN);
```
and so on

Reply:
Indeed. Nice idea.
I think I need a permission of @vittorioromeo to move from `typedef` to `using` in on;y one place of codebase.

Reply:
@Si1ver: Using `using` here is fine, I don't have a problem with in-flight PRs doing so. I will likely change all existing `typedef` to `using` before this PR is merged, anyway.

Reply:
Good. I'll follow your style then.